monitoring/statistics.h

```diff
@@ -135,4 +126,8 @@ inline void SetTickerCount(Statistics* statistics, uint32_t ticker_type,
   }
 }
 
+std::shared_ptr<Statistics> CreateDBStatistics() {
```

Comment:
Let's not using template, but pass ticker_max and histogram_max as optional param to `CreateDBStatistics` (adding a `DBStatisticsOptions` struct probably), and store the two enum max as member variable in `StatisticsImpl`.